CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 3.10)
+PROJECT(detours LANGUAGES C CXX)
+
+MACRO(GET_WIN32_WINNT version)
+    IF(WIN32 AND CMAKE_SYSTEM_VERSION)
+		SET(ver ${CMAKE_SYSTEM_VERSION})
+		STRING(REGEX MATCH "^([0-9]+).([0-9])" ver ${ver})
+		STRING(REGEX MATCH "^([0-9]+)" verMajor ${ver})
+		# Check for Windows 10, b/c we'll need to convert to hex 'A'.
+		IF("${verMajor}" MATCHES "10")
+			SET(verMajor "A")
+			STRING(REGEX REPLACE "^([0-9]+)" ${verMajor} ver ${ver})
+		ENDIF("${verMajor}" MATCHES "10")
+		# Remove all remaining '.' characters.
+		STRING(REPLACE "." "" ver ${ver})
+		# Prepend each digit with a zero.
+		STRING(REGEX REPLACE "([0-9A-Z])" "0\\1" ver ${ver})
+		SET(${version} "0x${ver}")
+    ENDIF()
+ENDMACRO()
+
+SET(LIB_MAJOR_VERSION "4")
+SET(LIB_MINOR_VERSION "0")
+SET(LIB_PATCH_VERSION "1")
+SET(LIB_VERSION_STRING "${LIB_MAJOR_VERSION}.${LIB_MINOR_VERSION}.${LIB_PATCH_VERSION}")
+
+# compile in release with debug info mode by default
+IF(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
+ENDIF()
+
+FILE(GLOB detours_src
+    ${CMAKE_CURRENT_LIST_DIR}/src/*.h
```

Comment:
Thank you for your suggestions!
Since I'm quite familiar with the Detours itself, I just use the script then it builded successfully. I will listing all these files here.
Then you're right, it is not nesscessary to add the header files. I'll remove the pattern.